dexlib2/src/main/java/org/jf/dexlib2/writer/DexWriter.java

```diff
@@ -533,7 +533,7 @@ private void writeClasses(@Nonnull DexDataStore dataStore, @Nonnull DexDataWrite
     }
 
     private boolean shouldWriteHiddenApiRestrictions() {
-        return hasHiddenApiRestrictions && opcodes.api >= 29;
```

Comment:
I'm pretty sure  this should stay as 29? Otherwise this will change the behavior for api 29.